ir/translation/literal.rs

```diff
@@ -169,13 +181,54 @@ impl FromTypeQLLiteral for NaiveDateTime {
     }
 }
 
-impl FromTypeQLLiteral for chrono::DateTime<chrono_tz::Tz> {
+impl FromTypeQLLiteral for chrono::DateTime<TimeZone> {
     type TypeQLLiteral = DateTimeTZLiteral;
 
     fn from_typeql_literal(literal: &Self::TypeQLLiteral) -> Result<Self, LiteralParseError> {
         let date = NaiveDate::from_typeql_literal(&literal.date)?;
         let time = NaiveTime::from_typeql_literal(&literal.time)?;
-        let tz: chrono_tz::Tz = todo!();
+        let tz = TimeZone::from_typeql_literal(&literal.timezone)?;
+        Ok(NaiveDateTime::new(date, time).and_local_timezone(tz).unwrap())
+    }
+}
+
+impl FromTypeQLLiteral for Duration {
+    type TypeQLLiteral = DurationLiteral;
+
+    fn from_typeql_literal(literal: &Self::TypeQLLiteral) -> Result<Self, LiteralParseError> {
+        let mut months = 0;
+        let mut days = 0;
+        let mut nanos = 0;
+
+        match literal {
+            DurationLiteral::Weeks(weeks) => days += 7 * u32::from_typeql_literal(weeks)?,
+            DurationLiteral::DateAndTime(date_part, time_part) => {
+                months +=
+                    MONTHS_PER_YEAR * date_part.years.as_ref().map(u32::from_typeql_literal).transpose()?.unwrap_or(0);
+                months += date_part.months.as_ref().map(u32::from_typeql_literal).transpose()?.unwrap_or(0);
+                days += date_part.days.as_ref().map(u32::from_typeql_literal).transpose()?.unwrap_or(0);
+
+                if let Some(time_part) = time_part {
+                    nanos += NANOS_PER_HOUR
+                        * time_part.hours.as_ref().map(u64::from_typeql_literal).transpose()?.unwrap_or(0);
+                    nanos += NANOS_PER_MINUTE
+                        * time_part.minutes.as_ref().map(u64::from_typeql_literal).transpose()?.unwrap_or(0);
+
+                    let seconds_decimal = (time_part.seconds.as_ref())
+                        .map(|seconds| parse_primitive::<Decimal>(&seconds.value))
+                        .transpose()?
+                        .unwrap_or_default();
+
+                    nanos += seconds_decimal.integer_part() as u64 * NANOS_PER_SEC;
+
+                    let nanos_decimal = (seconds_decimal - seconds_decimal.integer_part()) * NANOS_PER_SEC;
+                    debug_assert_eq!(nanos_decimal.fractional_part(), 0);
+                    nanos += nanos_decimal.integer_part() as u64;
```

Comment:
It would be cleaner to do `(seconds_decimal * NANOS_PER_SEC).integer_part()`, but that can overflow `i64` while still being in range for `u64`.